mistralrs-core/src/engine/mod.rs

```diff
@@ -69,6 +69,22 @@ impl Engine {
                 }
                 let logits = pipeline.forward(&scheduled.completion, false);
 
+                let logits = match logits {
+                    Ok(logits) => logits,
+                    Err(e) => {
+                        println!("COMPLETE - Model failed with error: {:?}", &e);
+                        for seq in scheduled.completion.iter_mut() {
+                            seq.responder()
+                                .send(Response::Error(e.to_string().into()))
+                                .unwrap();
+                            seq.set_state(SequenceState::Error);
+                        }
+                        Self::set_none_cache(&mut *pipeline);
+
+                        continue 'lp;
+                    }
+                };
+
```

Comment:
I don't like this code but I haven't given much thought to organization but rather behavior.

I think the waiting requests should also be set to errors? I'm not sure if `set_none_cache` alters them.

Reply:
No, the set_none_cache function just clears the model's cache. I don't think the waiting requests need to be set to Error, as they are not accessible in the loop by design.

I agree that this code is a bit repetitive, can you make this a macro in `utils`?